### PR TITLE
TF Keras local_test dataformat compatible test case updated

### PR DESCRIPTION
**Pull request by Dayananda-V**
TF Keras local_test dataformat compatible test case updated

### CODE REVIEW

File: tensorflow/python/keras/layers/local_test.py [hunk 1]
```diff
@@ -45,84 +48,84 @@ def test_locallyconnected_1d(self):
...
-                                  keras.layers.LocallyConnected1D,
-                                  **kwargs)
-              else:
-                testing_utils.layer_test(
-                    keras.layers.LocallyConnected1D,
-                    kwargs=kwargs,
-                    input_shape=(num_samples, num_steps, input_dim))
-
-  def test_locallyconnected_1d_regularization(self):
+          for implementation in [1, 2]:
```
Comment (fchollet) [inline tensorflow/python/keras/layers/local_test.py]: This can be made into a parameterization as well

File: tensorflow/python/keras/layers/local_test.py [hunk 2]
```diff
@@ -45,84 +48,84 @@ def test_locallyconnected_1d(self):
...
+  def test_locallyconnected_1d_regularization(self, data_format):
     num_samples = 2
     num_steps = 8
     input_dim = 5
     filter_length = 3
     filters = 4
-    for data_format in ['channels_first', 'channels_last']:
-      for padding in ['valid', 'same']:
-        for implementation in [1, 2]:
+    for padding in ['valid', 'same']:
```
Comment (fchollet) [inline tensorflow/python/keras/layers/local_test.py]: `padding` and `implementation` can be parameterized

File: tensorflow/python/keras/layers/local_test.py [hunk 3]
```diff
@@ -45,84 +48,84 @@ def test_locallyconnected_1d(self):
...
+            'implementation': implementation,
+            'padding': padding
+        }
+
+        if padding == 'same' and implementation == 1:
+          self.assertRaises(ValueError,
+                            keras.layers.LocallyConnected1D,
+                            **kwargs)
+        else:
+          with self.cached_session():
```
Comment (fchollet) [inline tensorflow/python/keras/layers/local_test.py]: Rather than using `self.cached_session`, please use the `run_all_keras_modes` decorator

**Review by fchollet — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: tensorflow/python/keras/layers/local_test.py (hunk 3 above)
Comment (Dayananda-V) [reply tensorflow/python/keras/layers/local_test.py]: run_all_keras_modes cause  absl.testing.parameterized.Duplicate Test NameError, so will use self.cached_session.

**Comment by Dayananda-V** [comment]
@fchollet 

Thanks for quicl review, please review the review comment fix changes.

**Comment by Dayananda-V** [comment]
Conflict changes checkin done under PR #27670
